### ISSUE TITLE
Level marked complete even when Azra is slain

### ISSUE THREAD

**Issue by IBwWG** [larsiusprime/tdrpg-bugs#722]
Unwelcome Guests on 0.5.7 just now, I failed, but it still gave me two blue stars.  I have the accessibility options cranked up, but I don't think that's one of them.

**Comment by larsiusprime** [comment]
I'm guessing this was caused by a recent regression -- Azra "summoning herself" when you use a saved battle layout.

1. Does it happen if you start a different battle, quit, start unwelcome guests, and fail?
2. Does it happen if you start unwelcome guests, immediately restart, and fail?

If answer to 1. is "no" and answer to 2. is "yes" then my hypothesis about the cause is correct. Total verification will proceed if this error does not re-occur in version 0.5.8, which supposedly fixes the Azra self-summon bug.

**Comment by IBwWG** [comment]
To be honest, I don't remember at this point.  I'll have to try again.

**Comment by IBwWG** [comment]
Ooh, fortunately I had one right at that point!

**Comment by IBwWG** [comment]
OK to answer neither of your questions, I couldn't reproduce this right now.  When I fight back a bit, then leave it so she dies, it doesn't crash, but I also don't get any stars.

**Comment by IBwWG** [comment]
Dang, was trying out 1) above and got null obj ref again.

**Comment by larsiusprime** [comment]
This is useful stuff nonetheless. That bug also looks like it's probably related two double-Azra's.

What happens is that when two Azras get summoned, one of them will be counted for the purposes of the map, one of them will be counted for the purposes of updating the sidebar UI, and one of them will be counted for the purpose of whether you won or not.

The above stack trace should be impossible to get with only one Azra on the map, because as soon as she's dead the battle should be *over* and no enemy should be able to reach the end of the map and attempt to damage her, which means that there's still a live Azra object around. However, in a double-Azra situation, it's possible one Azra gets killed, but the battle doesn't end, leaving a null object that can later get triggered.

0.5.7 still is vulnerable to the double-Azra bug. 0.5.8 fixes it, it's probably still building as we speak. I think there's nothing more to do here except wait and see if 0.5.8 fixes it.

**Comment by IBwWG** [comment]
OK 1) did not get any blue stars.

This wasn't what I was doing before (I'm not sure how I was losing before on Normal?) but anyway I play perfectly until the last two waves, then delete everyone, and then she dies without a crash.

**Comment by IBwWG** [comment]
The other thing is just now I'm on 800x600 non-HD sprites, and before I was one-under-1080p I think with HD sprites.  Maybe that has some effect.

**Comment by larsiusprime** [comment]
Okay I'm really sure this is a double Azra bug. Double-Azra would also explain the weird Azra HP sidebar non-updates.

**Comment by IBwWG** [comment]
OK cool.  So no need to try out the 2) option?

**Comment by larsiusprime** [comment]
Yeah, just wait for 0.5.8 before trying anything else.

**Comment by IBwWG** [comment]
Will do.  I don't quite understand why whether I fight back would make any difference, but oh well.

And before the "will do" here set in I was already trying with hi-res and HD, and got a null obj ref...after fighting back for only 4-5 waves before deleting everyone.  Same stack trace.

**Comment by IBwWG** [comment]
OK, at the very least it's not crashing if I just let it run on 0.6.0 without defenders.  :)

**Comment by IBwWG** [comment]
If by "start" [here](https://github.com/larsiusprime/tdrpg-bugs/issues/722#issuecomment-216288233) you mean just start, without placing defenders or waiting for a wave to arrive, then both cases 1 and 2 do not have the problem on 0.6.0.

**Comment by larsiusprime** [comment]
Yes, just starting and doing nothing is enough to set a new battle layout, that in the bugged version, would trigger "double Azra." So I'm calling this closed :)
